src/rapid_pro_to_engagement_db/rapid_pro_to_engagement_db.py

```diff
@@ -191,8 +201,19 @@ def sync_rapid_pro_to_engagement_db(rapid_pro, engagement_db, uuid_table, rapid_
             # De-identify the contact's full urn.
             contact = contacts_lut[run.contact.uuid]
             assert len(contact.urns) == 1, len(contact.urns)
-            contact_urn = contact.urns[0]
-            participant_uuid = _de_identify_contact_urn(contact_urn, uuid_table)
+            contact_urn = _normalise_and_validate_contact_urn(contact.urns[0])
+
+            if rapid_pro_config.uuid_filter is not None:
+                # If a uuid filter exists, then only add this message if the sender's uuid exists in the uuid table
+                # and in the valid uuids. The check for presence in the uuid table is to ensure we don't add a uuid
+                # table entry for people who didn't consent for us to continue to keep their data.
+                if not uuid_table.has_data(contact_urn) or uuid_table.data_to_uuid(contact_urn) not in valid_participant_uuids:
+                    log.info("Message not from a participant uuid in specified in the uuid filter; skipping")
```

Comment:
We should log and monitor both checks separetely.

Reply:
agreed, done